internal/weldr/api.go

```diff
@@ -385,13 +385,21 @@ func (api *API) openImageFile(composeId uuid.UUID, compose store.Compose) (io.Re
 	return reader, size, nil
 }
 
-// checkImageTypeDenylist checks the given ImageType name against the ImageType Denylist
-// provided to the API from configuration. If the given ImageType is not allowed the method
-// returns an `error`. Otherwise `nil` is returned.
-func (api *API) checkImageTypeDenylist(imageType string) error {
-	for _, deniedImgType := range api.imageTypeDenylist {
-		if imageType == deniedImgType {
-			return fmt.Errorf("image type denied by configuration: %q", imageType)
+// checkImageTypeDenylist checks the given ImageType and Distro names against
+//the distro-specific ImageType Denylist provided to the API from configuration.
+// If the given ImageType is not allowed the method returns an `error`.
+// Otherwise `nil` is returned.
+func (api *API) checkImageTypeDenylist(distroName, imageType string) error {
```

Comment:
Personally I'd make this return a bool. It's not really an error, you're just asking a question. I'd also name it something like `isImageTypeAllowed`

Reply:
I considered such name and returning bool. I'm honestly not sure now why I decided to go with returning error. I'll change it.

Reply:
I would change the name to `isImageTypeInDenyList` and make it return `bool`. The current name feels confusing to me.

Reply:
This is done.